e2e/main.e2e.spec.ts

```diff
@@ -40,69 +43,55 @@ test.describe.serial('Main App Test', () => {
   });
 
   test('should land on General page', async() => {
-    mainTitle = page.locator(mainTitleSelector);
+    const navPage = new NavPage(page);
 
-    await expect(mainTitle).toHaveText('Welcome to Rancher Desktop');
+    await expect(navPage.mainTitleSelector).toHaveText('Welcome to Rancher Desktop');
   });
 
   test('should start loading the background services and hide progress bar', async() => {
-    const progressBarSelector = page.locator('.progress');
+    const navPage = new NavPage(page);
 
-    // Wait until progress bar show up. It takes roughly ~60s to start in CI
-    await progressBarSelector.waitFor({ state: 'visible', timeout: 200_000 });
-    // Wait until progress bar be detached. With that we can make sure the services were started
-    await progressBarSelector.waitFor({ state: 'detached', timeout: 300_000 });
-    await expect(progressBarSelector).toBeHidden();
+    await navPage.progressBecomesReady();
+    await expect(navPage.progressBarSelector).toBeHidden();
   });
 
   test('should navigate to Kubernetes Settings and check elements', async() => {
-    const k8sMemorySliderSelector = '[id="memoryInGBWrapper"]';
-    const k8sCpuSliderSelector = '[id="numCPUWrapper"]';
-    const k8sPortSelector = '[data-test="portConfig"]';
-    const k8sResetBtn = '[data-test="k8sResetBtn"]';
-
-    await navigateTo('K8s');
-    // Collecting data from selectors
-    const k8sSettingsTitle = page.locator(mainTitleSelector);
-    const k8sMemorySlider = page.locator(k8sMemorySliderSelector);
-    const k8sCpuSlider = page.locator(k8sCpuSliderSelector);
-    const k8sPort = page.locator(k8sPortSelector);
-    const k8sResetButton = page.locator(k8sResetBtn);
+    const navPage = new NavPage(page);
+    const k8sPage = new K8sPage(page);
+
+    await navPage.navigateTo('K8s');
```

Comment:
API discussion:
Would it make sense to return the navigated-to page? Something like:
```ts
const navPage = new NavPage(page);
const k8sPage = await navPage.navigateTo('K8s');
```
We'd then have to implement it as something like:
```ts
const pageConstructors = {
  'K8s': K8sPage,
  'Integrations': WslPage,
  …
};
…
    async navigateTo(tab: keyof typeof pageConstructors) {
      await this.page.click(`.nav li[item="/${ tab }"] a`);
      await this.page.waitForNavigation({ url: `**/${ tab }`, timeout: 60_000 }),
      return new (pageConstructors[tab])(this.page);
    }
```

Reply:
Thanks for sharing the idea.

That's is useful to perform browser level control, as you have access to the page property. 
AFAIK playwright will keep the `page` property available after this [process](https://playwright.dev/docs/api/class-electronapplication#electron-application-first-window) : 
`page = await electronApp.firstWindow();`

`page` will be globally available across all test cases.
Returning `page` from each tab, I cannot have access to the selector from the POM page, requiring another instance t `const k8sPageItems =  new K8sPage(page);`

The way that I've implemented the `navigateTo()`, it's basically to simulate how a user will perform, navigating to the tab and using the selector `.nav`.

@mook-as 
I'm not a TS guru yet, but please, share your thoughts :)

Reply:
Hmm, let me rephrase what you said to ensure I understand you correctly:
You're saying that we would want to have access to a page without navigating to it first (because we expect it to already be there), is that correct? In that case, I'd view that as an optimization and it's better to do redundant actions in the test so that when editing the test we don't need to keep the context of related tests in our head, specially in cases like deleting a test.

> Returning `page` from each tab, I cannot have access to the selector from the POM page, requiring another instance t `const k8sPageItems = new K8sPage(page);`

Could you clarify which _selector_ you're referring to here? Given that the argument passed to `new (pageConstructors[tab])(this.page)` is just the same one that was passed to `new NavPage(page);`, I don't see how it's different from the test code calling `new K8sPage(page);` directly.  The test would then call things like `await expect(k8sPage.memorySlider).toBeVisible();` as you do now.

Thanks! Please let me know if a scratch branch to demonstrate what I'm thinking of would make things clearer.

Reply:
> Hmm, let me rephrase what you said to ensure I understand you correctly: You're saying that we would want to have access to a page without navigating to it first (because we expect it to already be there), is that correct? In that case, I'd view that as an optimization and it's better to do redundant actions in the test so that when editing the test we don't need to keep the context of related tests in our head, specially in cases like deleting a test.

Nope,
We need to navigate to the page in order to load the page elements, otherwise, we can't check the components.

> Could you clarify which selector you're referring to here? Given that the argument passed to new (pageConstructors[tab])(this.page) is just the same one that was passed to new NavPage(page);, I don't see how it's different from the test code calling new K8sPage(page); directly. The test would then call things like await expect(k8sPage.memorySlider).toBeVisible(); as you do now.

I meant the HTML components to be asserted:
```typescript
      this.mainTitle = page.locator('[data-test="mainTitle"]');
      this.progressBar = page.locator('.progress');
```

I've created this [branch](https://github.com/evertonlperes/rancher-desktop/tree/sandbox-mark-thoughts) and implemented your suggestions.

Let's discuss a bit more and really appreciate your support :)